### ISSUE TITLE
[BUG, CVE-2018-11205, HDFFV-10479] 

### ISSUE THREAD

**Issue by e4t** [HDFGroup/hdf5#2250]
**Description**
The [reproducer](https://github.com/TeamSeri0us/pocs/tree/master/hdf5) of CVE-2018-11205 triggers an out of bound read in memcpy() leading to a SEGV: The crash happens when reading compact chunked data. The hdf5 file seems to specify a selection whose size exceeds the size of the chunked data.

**Expected behavior**
It should be verified that selected data lies within the chunk before attempting to read.

**Platform**
  - 1.10.7 and earlier
 - openSUSE Leap 15.4/ SLE 15 SP4 / openSUSE Tumbleweed
 - gcc7
 - Autotools
 -  config --enable-fortran --enable-unsupported --enable-hl --enable-shared --enable-threadsafe --enable-build-mode=production --enable-cxx --with-pthread

**Comment by fortnern** [comment]
This appears to be fixed in newer versions of HDF5 (tested 1.10 branch and develop). @e4t is there anything to do here?

**Comment by e4t** [comment]
> This appears to be fixed in newer versions of HDF5 (tested 1.10 branch and develop). @e4t is there anything to do here?

@fortnern : I can see this fixed in the head of the 1.14 branch (commit cac778a4c), however, I can still reproduce the segfault with the head of the 1.10 branch (commit 779db127). I cannot test 1.12 ATM as the head seems to require autoconf v. 2.71.
Not sure if you plan to include all CVE fixes into 1.10 as well. The changes that fixed CVE-2018-11205 in the end were different than what I had proposed in #2251 (as stated, I was not convinced of it either). I cannot find a reference to the CVE in the changelog so I assume this was fixed as part of a larger restructuring.

Output 1.14:
```
$ h5dump ./1093663/H5VM_memcpyvv-H5VM.c_1626-oob_read
HDF5 "./1093663/H5VM_memcpyvv-H5VM.c_1626-oob_read" {
GROUP "/" {
   DATASET "Compressed_Data" {
      DATATYPE  H5T_STD_I32BE
      DATASPACE  SIMPLE { ( 100, 20 ) / ( 100, 20 ) }
      DATA {h5dump error: unable to print data

      }
   }
}
}
```
Output 1.10:
```
h5dump ./1093663/H5VM_memcpyvv-H5VM.c_1626-oob_read
HDF5 "./1093663/H5VM_memcpyvv-H5VM.c_1626-oob_read" {
GROUP "/" {
   DATASET "Compressed_Data" {
      DATATYPE  H5T_STD_I32BE
      DATASPACE  SIMPLE { ( 100, 20 ) / ( 100, 20 ) }
Segmentation fault (core dumped)
```

**Comment by derobins** [comment]
Definitely fixed in 1.14.3, 1.12.3, and 1.10.11
